launcher/modplatform/atlauncher/ATLPackManifest.cpp

```diff
@@ -232,4 +259,25 @@ void ATLauncher::loadVersion(PackVersion & v, QJsonObject & obj)
         auto configsObj = Json::requireObject(obj, "configs");
         loadVersionConfigs(v.configs, configsObj);
     }
+
+    if (obj.contains("colours")) {
+        auto colourObj = Json::requireObject(obj, "colours");
+
+        for (const auto &key : colourObj.keys()) {
+            v.colours[key] = Json::requireString(colourObj.value(key), "colour");
+        }
+    }
```

Comment:
I think you can use `Json::ensureObject`, which will return an empty object if something fails. Then you can probably skip the `if`

```suggestion
    auto colourObj = Json::ensureObject(obj, "colours");

    for (const auto &key : colourObj.keys()) {
        v.colours[key] = Json::requireString(colourObj.value(key), "colour");
    }
```

Reply:
This matches how I've written the rest of that file, as it is the style I prefer. I can make this change if desired though, it should work :)